test/integration/network_reserved_ips_test.go

```diff
@@ -2,150 +2,199 @@ package integration
+	if fetchErr != nil {
+		t.Errorf("Error getting reserved IP address: %v", fetchErr)
+	}
 
-		// Verify the IP has been deleted
-		if reserveIP != nil {
-			verifyDeletedIP, fetchDeletedIpErr := client.GetReservedIPAddress(context.Background(), reserveIP.Address)
-			if fetchDeletedIpErr != nil {
-				t.Logf("Error getting ipaddress, expected struct, got %v and error %v", verifyDeletedIP, fetchDeletedIpErr)
-			}
+	if fetchedIP == nil {
+		t.Errorf("Retrieved reserved IP is nil")
+	}
 
-			verifyDeletedFromList, verifyDeletedFromListIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-			if verifyDeletedFromListIpErr != nil {
-				t.Logf("Error listing ipaddresses, expected struct, got error %v", verifyDeletedFromListIpErr)
-			} else {
-				if len(verifyDeletedFromList) < len(verifyList) {
-					t.Log("IP address deletion confirmed", verifyDeletedFromList)
-				} else {
-					t.Errorf("Verification - Failed")
-				}
-			}
-		}
+	// Verify the list of IPs has increased
+	verifyList, verifyErr := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+	if verifyErr != nil {
+		t.Fatalf("Error listing IP addresses after reservation: %v", verifyErr)
+	}
+
+	if len(verifyList) != initialCount+1 {
+		t.Errorf("Expected IP count to increase by 1, got %d, want %d", len(verifyList), initialCount+1)
+	}
+
+	// Delete the reserved IP
+	delErr := client.DeleteReservedIPAddress(context.Background(), resIP.Address)
+	if delErr != nil {
+		t.Fatalf("Error deleting reserved IP address: %v", delErr)
+	}
+
+	// Verify the IP has been deleted
+	_, fetchDelErr := client.GetReservedIPAddress(context.Background(), resIP.Address)
+	if fetchDelErr == nil {
+		t.Errorf("Expected error when fetching deleted IP, got nil")
+	}
+
+	verifyDelList, verifyDelErr := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+	if verifyDelErr != nil {
+		t.Fatalf("Error listing IP addresses after deletion: %v", verifyDelErr)
+	}
+
+	if len(verifyDelList) != initialCount {
+		t.Errorf("Expected IP count to return to initial count, got %d, want %d", len(verifyDelList), initialCount)
 	}
 }
 
-// TestReservedIPAddresses_ListIPAddressesVariants tests various scenarios for listing IP addresses
+// TestReservedIPAddresses_ListIPAddressesVariants tests filters for listing IP addresses
 func TestReservedIPAddresses_ListIPAddressesVariants(t *testing.T) {
 	client, teardown := createTestClient(t, "fixtures/TestReservedIPAddresses_ListIPAddressesVariants")
 	defer teardown()
 
-	filter := ""
-	ipList, listIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-	if listIpErr != nil {
-		t.Logf("Error listing ipaddresses, expected struct, got error %v", listIpErr)
-	} else {
-		if len(ipList) == 0 {
-			t.Logf("The customer has not reserved an IP %v", ipList)
+	// Reserve two IP addresses in us-east region
+	reservedIPs := make([]string, 2)
```

Comment:
It's good practice to store constants in a variable since it clarifies its meaning and intention and makes it easier to modify (are all the 2's you used below related to ips?). For example, `expected_ips := 2`.

Reply:
Yes all the 2's are related to the number of IP's that this test will be reserving and then listing. Created a constant instead and used it everywhere.